src/columns.jl

```diff
@@ -314,7 +314,14 @@ end
 
 function permute!(c::Columns, p::AbstractVector)
     for v in c.columns
-        copy!(v, v[p])
+        if isa(v, StringArray{String})
+            ws = convert(StringArray{WeakRefString{UInt8}}, v)
+            permute!(ws, p)
```

Comment:
Ideally this should be the implementation of this in WeakRefStrings.

Reply:
PR opened: https://github.com/JuliaData/WeakRefStrings.jl/pull/39.
Will remove from here once that's merged.